paddle/fluid/operators/mkldnn/mul_mkldnn_op.cc

```diff
@@ -379,19 +321,19 @@ inner_product_forward GetMulPrimitive(const MKLDNNDeviceContext &dev_ctx,
                                       const Tensor *input_x,
                                       const Tensor *input_y, Tensor *output,
                                       const mkldnn::engine &mkldnn_engine) {
-  bool enable_quant =
+  constexpr bool is_int8 =
       std::is_same<XT, int8_t>::value || std::is_same<XT, uint8_t>::value;
   bool force_fp32_output = ctx.Attr<bool>("force_fp32_output");
 
-  if (enable_quant && !force_fp32_output) {
+  if (is_int8 && !force_fp32_output) {
     return GetPrimitiveFactory<XT, YT, int8_t>(dev_ctx, ctx, input_x, input_y,
-                                               mkldnn_engine, enable_quant)
-        ->CreateMulPrimitive(input_x, input_y, output, ctx);
+                                               mkldnn_engine, is_int8)
```

Comment:
I think adding `is_int8` as a `static constexpr bool` field in the `MulPrimitiveFactory` would save us passing the `is_int8` argument to a few methods.